packages/svelte-materialify/src/components/Textarea/Textarea.svelte

```diff
@@ -30,33 +30,34 @@
   export let style = null;
   export let textarea = null;
 
-  let labelActive = !!placeholder || value;
+  let focused = false;
+  $: labelActive = !!placeholder || value || focused;
   let errorMessages = [];
 
-  function checkRules() {
+  export function validate() {
     errorMessages = rules.map((r) => r(value)).filter((r) => typeof r === 'string');
     if (errorMessages.length) error = true;
     else {
       error = false;
     }
+    return error;
   }
 
   function onFocus() {
-    labelActive = true;
+    focused = true;
   }
 
   function onBlur() {
-    if (!value && !placeholder) labelActive = false;
-    if (validateOnBlur) checkRules();
+    focused = false;
+    if (validateOnBlur) validate();
   }
 
   function clear() {
     value = '';
-    if (!placeholder) labelActive = false;
   }
 
   function onInput() {
-    if (!validateOnBlur) checkRules();
+    if (!validateOnBlur) validate();
     if (autogrow) {
       textarea.style.height = 'auto';
       textarea.style.height = `${textarea.scrollHeight}px`;
```

Comment:
The only thing I'm concerned is that is the value is dynamically changed and gets too big (and `autogrow` is activated), it might not grow automatically